app/services/PayPalNvpService.scala

```diff
@@ -56,9 +57,14 @@ class PayPalNvpService(apiConfig: PayPalConfig, wsClient: WSClient) extends Touc
 
   // Takes an NVP response and retrieves a given parameter as a string.
   private def retrieveNVPParam(response: QueryString, paramName: String) =
-    response.paramMap(paramName).head
+    response.paramMap(paramName).headOption match {
+      case None =>
+        SafeLogger.warn(s"Parameter $paramName was missing from the NVP response - $response")
+        None
+      case Some(value) => Some(value)
```

Comment:
not sure if it's the case here, but for deterministicness sake I tend to avoid headOption if I expect exactly one item, usually I do a pattern match on a head :: Nil and then send everything else to the bin

Reply:
also you are doing a pattern match and returning exactly the same thing which is a bit surprising - could just put the overall headOption into a val and have a `if (val.isEmpty)  logger.warn` and then return the val at the end